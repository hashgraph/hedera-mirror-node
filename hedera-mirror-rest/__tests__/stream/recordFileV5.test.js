/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

import RecordFileV5 from '../../stream/recordFileV5';
import testUtils from './testUtils';

describe('unsupported record file version', () => {
  testUtils.testRecordFileUnsupportedVersion([1, 2, 3, 4, 6], RecordFileV5);
});

describe('canCompact', () => {
  const testSpecs = [
    [1, false],
    [2, false],
    [3, false],
    [4, false],
    [5, true],
    [6, false],
  ];

  testUtils.testRecordFileCanCompact(testSpecs, RecordFileV5);
});

describe('from v5 buffer or compact object', () => {
  testUtils.testRecordFileFromBufferOrObj(5, RecordFileV5, true, true);
});
