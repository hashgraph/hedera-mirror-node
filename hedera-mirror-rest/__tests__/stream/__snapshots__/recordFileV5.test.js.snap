// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`from v5 buffer or compact object check individual field from v5 buffer - toCompactObject - ["0.0.1-123-123456789"] 1`] = `"Transaction 0.0.1-123-123456789 not found in the successful transactions map"`;

exports[`from v5 buffer or compact object check individual field from v5 compact object - toCompactObject - ["0.0.1-123-123456789"] 1`] = `"Transaction 0.0.1-123-123456789 not found in the successful transactions map"`;

exports[`from v5 buffer or compact object end running hash mismatch 1`] = `"End object running hash mismatch"`;

exports[`from v5 buffer or compact object truncated v5 buffer 1`] = `"Error reading byte array, expect 48-byte data got 47-byte"`;

exports[`from v5 buffer or compact object v5 buffer with extra data 1`] = `"Extra data discovered in record file"`;

exports[`unsupported record file version create object from version 1 1`] = `"Unsupported record file version, expect 5"`;

exports[`unsupported record file version create object from version 2 1`] = `"Unsupported record file version, expect 5"`;

exports[`unsupported record file version create object from version 3 1`] = `"Unsupported record file version, expect 5"`;

exports[`unsupported record file version create object from version 4 1`] = `"Unsupported record file version, expect 5"`;

exports[`unsupported record file version create object from version 6 1`] = `"Unsupported record file version, expect 5"`;
